### PR TITLE
Addressing Underspecification in Machine Learning

### PR DESCRIPTION
**Pull request by collins-a**
Article as index.md.
Hero image as hero.jpg

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
this fixes #756

**Comment by hectorkambow** [comment]
Plagiarism check done

File: articles/underspecification/index.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+### The problem
+
+Underspecification threatens the credibility of models. The reliability of the process used to train models to perform in the real world as they do in training has been called into question. We usually expose a model to a lot of examples and test it on unseen data. This is the typical training process. If such models do well in testing, we consider them to be good models. Yet, this might not be the case.
+
+Consider a scenario where we are training many similar models on the same data. These models may differ in tiny ways, such as the number of training runs or the order in which training data is put in. These nuances between the models rarely impact performance. This is why these variations are often overlooked while evaluating models. Yet, as per the findings of this [paper](https://arxiv.org/pdf/2011.03395.pdf), they may have a significant impact on models' real-world performance.
```
Comment (lalith1403) [inline articles/underspecification/index.md:29]: model's real-world performance.

File: articles/underspecification/index.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+The three types of stress tests include:
+
+**Stratified Performance Evaluations**. An [inductive bias](http://inductivebias.com/Blog/what-is-inductive-bias/) refers to a set of assumptions that a learner uses to make predictions given never-before encountered inputs. Stratified performance evaluations test whether a predictor encodes inductive biases that give similar performances across different dataset classes. These tests consider sub-groups of observations to identify inequalities.
```
Comment (lalith1403) [inline articles/underspecification/index.md:51]: The second sentence needs some more explanation. How do the inductive biases get tested for?

File: articles/underspecification/index.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+**Shifted Performance Evaluations**. Shifted performance evaluations evaluate whether the average performance generalizes when the test-set distribution is different from the training set distribution. This is done by spawning a made-up test distribution. This made-up distribution is created by deliberately causing perturbations on data. The perturbations are to keep track of the behavior of the model.
+
+**Contrastive Evaluations**. These tests manipulate an input observation to compute the significance of the effect on the output. These evaluations support localized analysis of specific inductive biases. They are performed on an example/observation as opposed to data distribution. The goal is to check whether modifying a given input causes the model's output to change in unanticipated ways.
```
Comment (lalith1403) [inline articles/underspecification/index.md:55]: Does this come under explainable AI? If I can figure out which aspect contributes to the output, then can we explore this test for all models? What is the catch here?

File: articles/underspecification/index.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+These stress tests are of great use. They highlight cases where models are unsuccessful at generalizing as needed for deployment. You can learn more about these three stress tests by reading this [paper](https://arxiv.org/pdf/2011.03395.pdf).
+
+#### Improving Training and Testing Processes
+
+Underspecification presents an argument that suggests that the way we train our models needs a lot of improvement. As a response to this argument, there are a couple of approaches worth considering. First, we could ensure that we have a test-set that represents a real-world environment. We could also make sure that our training data is as close to a real-world distribution as possible. Let's expound on the two approaches.
+
+##### Test Set that Reflects Deployment Environment
```
Comment (lalith1403) [inline articles/underspecification/index.md:63]: Test set reflecting deployment environment

**Review by lalith1403 — CHANGES_REQUESTED** [review]
I learned a lot from this article about underspecification. Amazing job, @collins-a. A few minor changes and this should be ready to go.

**Comment by collins-a** [comment]
@lalith1403 hey, I'm glad you found the article informative. I've made the needed minor changes. I hope they suffice. Catch you next article!

**Pull request merged**